sdk/python/feast/infra/offline_stores/file_source.py

```diff
@@ -182,17 +183,31 @@ def __init__(
         file_format: Optional[FileFormat],
         file_url: Optional[str],
         s3_endpoint_override: Optional[str],
+        uri: Optional[str],
     ):
         """
         FileOptions initialization method
 
         Args:
             file_format (FileFormat, optional): file source format eg. parquet
-            file_url (str, optional): file source url eg. s3:// or local file
-            s3_endpoint_override (str, optional): custom s3 endpoint (used only with s3 file_url)
+            file_url (str, optional): [DEPRECATED] file source url eg. s3:// or local file
+            s3_endpoint_override (str, optional): custom s3 endpoint (used only with s3 uri)
+            uri (str, optional): file source url eg. s3:// or local file
+
         """
         self._file_format = file_format
+        if file_url:
+            warnings.warn(
+                (
+                    "The option to pass a file_url parameter to FileOptions is being deprecated. "
+                    "Please pass the file url to the uri parameter instead. The parameter will be deprecated in Feast 0.23"
+                ),
+                DeprecationWarning,
+            )
         self._file_url = file_url
+        self._uri = uri
+        if not self._uri:
+            self._uri = self._file_url
```

Comment:
you can do `self._uri = uri or file_url` and probably not store `self._file_url` at all

Reply:
Great catch! thanks!